llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -553,28 +801,36 @@ void processFunction(Function &F, FunctionToAspectsMapTy &FunctionToUsedAspects,
       getAspectsFromType(F.getReturnType(), TypesWithAspects);
   for (const auto &Aspect : RetTyAspects)
     if (!FP64ConvEmu || (Aspect != FP64Aspect) ||
-        !hasDoubleType(F.getReturnType()))
-      FunctionToUsedAspects[&F].insert(Aspect);
+        !hasDoubleType(F.getReturnType())) {
+      if (FunctionIsUnconditional)
+        FunctionToUsedAspects[&F].insert(Aspect);
+      if (FunctionIsConditional)
+        FunctionToConditionallyUsedAspects[&F].insert(Aspect);
```

Comment:
`FunctionIsConditional` is actually initialized from `Funcs.CalledConditionally`. This raises a couple of questions:
- could the same function be called both conditionally and unconditionally?
- even if a function is called unconditionally, doesn't it still mean that the aspect *it* uses are unconditional? because they are always used by that function - at least those which are "inherited" from return type and argument types. Caller may or may not use those aspect, but for the function itself they do seem to be unconditional enough

The second point makes me think that the only time we should add something into `FunctionToConditionallyUsedAspects` is when we handle a conditional call instruction.